### DISCUSSION TITLE
clang工具链在windows 上显示颜色

### DISCUSSION THREAD

**Discussion by ODtian** [xmake-io/xmake-vscode#260] [Ideas]
clang在windows上默认用的是console api，导致如果不是直接调用无法显示颜色，而windows在10之后加入了ansi颜色，clang也可以选择在windows平台下使用ansi颜色，但不是默认开启。如果可以默认开启ansi颜色就好了

**Comment by waruqi** [comment]
是否支持 color output 是自动探测的，https://github.com/xmake-io/xmake/blob/e1d5f432ab3d0fac43ee6a4852efe7b99f6ef428/xmake/modules/core/tools/gcc.lua#L566

如果当前 clang 支持，就会自动输出 color 。。

但是当前 win terminal 是否支持 color code ，也有自动检测，检测无法支持的，就会禁用。。。

**Reply by ODtian** [reply]
clang在windows平台上会检测是不是console环境，如果是的话会使用控制台api输出颜色，因为win10之前windows没有ansi颜色，而如果不是，比如在cmake和xmake这样的构建工具里调用就不会输出颜色，且无论有没有开启颜色选项都不会输出颜色。

win10之后clang支持在windows上输出ansi颜色，但不是默认行为，需要`-f-ansi-escape-codes`手动开启。cmake也有这个问题https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8900

**Reply by waruqi** [reply]
？那现在啥问题呢？我看终端上，clang 的 color output 不是正常的么。。

是要在非终端环境输出 color ?

**Reply by ODtian** [reply]
现在我的项目里clang默认配置确实没法显示颜色，想要显示颜色必须添加`add_cxflags("-fansi-escape-codes")`
![image](https://github.com/xmake-io/xmake-vscode/assets/40318251/cee05b1b-52cf-4819-a26b-cbb240427bc2)
![image](https://github.com/xmake-io/xmake-vscode/assets/40318251/b50d8616-edb9-4243-bbee-d542f17ee03f)
上图为vscode里的终端，下图为windows terminal，没有颜色的是没加`add_cxflags("-fansi-escape-codes")`，有颜色是加了的

**Comment by waruqi** [comment]
再试试 https://github.com/xmake-io/xmake/pull/4875

```console
xmake update -s github:xmake-io/xmake#clang
```

**Reply by ODtian** [reply]
ok了，感谢
